src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java

```diff
@@ -854,34 +822,75 @@ public boolean shouldSkipMonitoredTripCheck(boolean persist) throws Exception {
             // It's been about an hour since the last check. Do not skip.
             int overHourCheckThresholdMinutes = 60;
             if (minutesSinceLastCheck >= overHourCheckThresholdMinutes) {
-                // TODO: Change log level.
-                LOG.info("Trip not checked in at least an {} minutes. Checking.", overHourCheckThresholdMinutes);
+                LOG.debug("Trip not checked in at least an {} minutes. Checking.", overHourCheckThresholdMinutes);
                 return false;
             }
         } else {
             // It's less than an hour until the trip time, start more frequent trip checks (about every 15 minutes).
             if (minutesSinceLastCheck >= 15) {
                 // Last check was more than 15 minutes ago. Check. (approx. 4 checks per hour).
-                // TODO: Change log level.
-                LOG.info("Trip happening soon. Checking.");
+                LOG.debug("Trip happening soon. Checking.");
                 return false;
             }
             // If the trip starts within 30 minutes, check the trip every minute (assuming the loop runs every minute).
             int checkEveryMinuteThresholdMinutes = 30;
             if (minutesUntilTrip <= checkEveryMinuteThresholdMinutes) {
-                // TODO: Change log level.
-                LOG.info("Trip happening within {} minutes. Checking every minute.", checkEveryMinuteThresholdMinutes);
+                LOG.debug("Trip happening within {} minutes. Checking every minute.", checkEveryMinuteThresholdMinutes);
                 return false;
             }
         }
         // TODO: Check that journey state is not flagged
         // TODO: Check last notification time.
         // Default to skipping check.
-        // TODO: Change log level.
-        LOG.info("Trip criteria not met to check. Skipping.");
+        LOG.debug("Trip criteria not met to check. Skipping.");
         return true;
     }
 
+    /**
+     * Calculate target time for the next trip plan request find the next possible day the trip is active by
+     * initializing the appropriate target time. Start by checking today's date at the earliest in case the user has
+     * paused trip monitoring for a while.
+     */
+    private void setTargetDateTime() {
```

Comment:
Can you create function that computes and returns the date/time of the next trip?

Reply:
@binh-dam-ibigroup When you say function are you thinking Supplier?

Reply:
`public ZonedDateTime computeTargetDateTime() { ... }` - Nothing fancy. Then, in `shouldSkipMonitoredTripCheck`, write:
```
targetZoneDateTime = computeTargetDateTime();
```